client.go

```diff
@@ -335,19 +335,29 @@ func (c *client) UploadFile(path string, contents io.ReadCloser) error {
 	if err != nil {
 		return fmt.Errorf("sftp: problem creating %s: %w", path, err)
 	}
-	defer fd.Close()
 
 	n, err := io.Copy(fd, contents)
 	if err != nil {
 		return fmt.Errorf("sftp: problem copying (n=%d) %s: %w", n, path, err)
 	}
 
+	if err := fd.Sync(); err != nil {
+		// Skip sync if the remote server doesn't support it
+		if !strings.Contains(err.Error(), "SSH_FX_OP_UNSUPPORTED") {
+			return fmt.Errorf("sftp: problem with sync on %s: %v", path, err)
+		}
+	}
+
 	if !c.cfg.SkipChmodAfterUpload {
 		if err = fd.Chmod(0600); err != nil {
 			return fmt.Errorf("sftp: problem chmod %s: %w", path, err)
 		}
 	}
 
+	if err := fd.Close(); err != nil {
+		return fmt.Errorf("sftp: closing %s after writing failed: %w", path, err)
```

Comment:
Are there any errors that should be ignored here? Some Go libs return errors on close that aren't actually errors, like when you close an http server.

Reply:
Not that I've ran into. We've been [requiring no such errors](https://github.com/moov-io/achgateway/blob/v0.23.1/internal/upload/sftp.go#L346-L348) for a few years and haven't seen a false positive. The `sftp` package we use [just sends a close packet](https://github.com/pkg/sftp/blob/v1.13.6/client.go#L619-L637), which should be supported.